### PR TITLE
Macos support

### PR DESCRIPTION
**Pull request by voicubabiciu**
Added support for macOS platform and also implemented the `fetchPlace` function

Closes [26](https://github.com/matanshukry/flutter_google_places_sdk/issues/26)

### CODE REVIEW

File: flutter_google_places_sdk_macos/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: flutter_google_places_sdk_macos
+description: The macos implementation of Flutter plugin for google places sdk
+publish_to: none
+
```
Comment (matanshukry) [inline flutter_google_places_sdk_macos/pubspec.yaml:4]: where's the version? please set it to `0.1.0+0`

Comment (matanshukry) [inline flutter_google_places_sdk_macos/pubspec.yaml:4]: add
`homepage: homepage: https://github.com/matanshukry/flutter_google_places_sdk/tree/master/flutter_google_places_sdk_macos`

File: flutter_google_places_sdk_macos/analysis_options.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# This file configures the analyzer, which statically analyzes Dart code to
```
Comment (matanshukry) [inline flutter_google_places_sdk_macos/analysis_options.yaml:1]: please follow the other files (sadly they're not standarized/imports, but hopefully I'll do it one day).
e.g.
```
linter:
  rules:
    - public_member_api_docs
    - package_api_docs
analyzer:
  errors:
    invalid_annotation_target: ignore
```

File: flutter_google_places_sdk_macos/lib/flutter_google_places_sdk_macos.dart [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+import 'package:flutter_google_places_sdk_platform_interface/flutter_google_places_sdk_platform_interface.dart'
+    as inter;
+import 'package:http/http.dart' as http;
+import 'package:latlong2/latlong.dart' as latlnglib;
+
+import 'types/types.dart';
+
+final _kLogPrefix = 'flutter_google_place_sdk_macos_plugin :: WARN -';
+
+/// Web implementation plugin for flutter google places sdk
```
Comment (matanshukry) [inline flutter_google_places_sdk_macos/lib/flutter_google_places_sdk_macos.dart:16]: `Web` -> `macOS`

File: flutter_google_places_sdk_macos/lib/types/place_field.dart [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:flutter_google_places_sdk_platform_interface/flutter_google_places_sdk_platform_interface.dart'
+    as inter;
+part 'place_field.g.dart';
+
+/// Used to specify which place data types to return.
+///
+/// Ref: https://developers.google.com/maps/documentation/places/android-sdk/reference/com/google/android/libraries/places/api/model/Place.Field
```
Comment (matanshukry) [inline flutter_google_places_sdk_macos/lib/types/place_field.dart:8]: that's ref for the android documentation

**Review by matanshukry — COMMENTED** [review]
I didn't finish it, but it's quite long so I'll try to do it in phases.

**Comment by matanshukry** [comment]
@voicubabiciu since the impl is the same as windows impl, they should be a single one rather than duplicated.

I did the work of extracting the windows package to another one ("http" now), and also changed the window to use it.

What should be done now then:
1. Impl the `fetchPlace` into the `..._http` library
2. Create a macos package, much like the window one, that uses the http one.
3. Add the macos to the default package impl
4. Optional: upgrade window to use the new `..._http` library after implementing the `fetchPlace`

Note that each point need to be a separate commit (hence separate PR).

@voicubabiciu let me know if you want to do it - otherwise I'll just implement it.
And thanks for your contribution btw!

**Comment by voicubabiciu** [comment]
@matanshukry Hello, sorry but right now I don't have enough time to work on it. Feel free to implement it.

**Comment by matanshukry** [comment]
@voicubabiciu done, implemented.

I don't have personal mac to test this - but the implementation is pretty straightforward - so 🤞

**Pull request closed without merging**